src/proposals/proposal.cpp

```diff
@@ -45,22 +48,62 @@ namespace hypha
         // the DHO also links to the document as a proposal, another graph EDGE
         Edge::write(m_dao.get_self(), proposer, root, proposalNode.getHash(), common::PROPOSAL);
 
+        // Sets an empty tally
+        updateVoteTally(proposalNode, proposer);
+
         postProposeImpl(proposalNode);
 
         return proposalNode;
     }
 
     void Proposal::postProposeImpl(Document &proposal) {}
 
+    void Proposal::vote(const eosio::name &voter, const std::string vote, Document& proposal)
+    {
+        proposal.getContentWrapper().getOrFail(BALLOT_OPTIONS, vote, "Invalid vote");
+
+        // Fetch vote power
+        trailservice::trail::voters_table v_t(voter, voter.value);
```

Comment:
Is logic correct?  or should be 
```
name trailContract = m_dao.getSettingOrFail<eosio::name>(TELOS_DECIDE_CONTRACT)
trailservice::trail::voters_table v_t(trailContract, voter.value);
```

Reply:
The comment is correct.  The voters table is actually stored inside the trailContract, so that must be the first parameter when instantiating the table.